### DISCUSSION TITLE
SMTP not configured

### DISCUSSION THREAD

**Discussion by tpetersvw** [dani-garcia/vaultwarden#3216] [Q&A] [answered]
Hi Guys,

I've set up a vaultwarden environment with with the following docker compose file. Unfortunately the SMTP settings are completely missing from the web interface. When i add a user to the org (+click resend invitation), the logs provide me with the following error:

[2023-02-02 09:26:34.641][vaultwarden::api::core::organizations][ERROR] SMTP is not configured.

Am i missing some sort of configuration to enable the SMTP features?

docker-compose.yaml:

version: '4'

services:
  vaultwarden:
    image: vaultwarden/server:latest
    container_name: vaultwarden
    restart: always
    environment:
      WEBSOCKET_ENABLED: "true"  # Enable WebSocket notifications.
    volumes:
      - ./vw-data:/data

  caddy:
    image: caddy:2
    container_name: caddy
    restart: always
    ports:
      - 80:80  # Needed for the ACME HTTP-01 challenge.
      - 443:443
    volumes:
      - ./certs:/etc/letsencrypt/archive
      - ./Caddyfile:/etc/caddy/Caddyfile:ro
      - ./caddy-config:/config
      - ./caddy-data:/data
      - caddy-logs:/logs
    environment:
      DOMAIN: "https://vaultwarden.bla.bla"  # Your domain.
      EMAIL: "administrator@bla.bla"                # The email address to use for ACME registration.
      LOG_FILE: "/data/access.log"
      SMTP_DEBUG: true
      SMTP_HOST: mail.bla.bla
      SMTP_FROM: "vaultwarden@bla.bla"
      SMTP_PORT: 25
      SMTP_SECURITY: starttls
      SMTP_USERNAME: "bla@bla.bla"
      SMTP_PASSWORD: "bla"
volumes:
  vw-data:
  caddy-config:
  caddy-data:
  caddy-logs:

EDIT:

Turns out i need to enable the admin page first:

https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page

I've added a token but i'm still getting the error: on /admin

The admin panel is disabled, please configure the 'ADMIN_TOKEN' variable to enable it

**Comment by stefan0xC** [comment, marked as answer]
It looks like you have added the environment variables to the caddy service and not to vaultwarden.

**Reply by tpetersvw** [reply]
o man, i'm such an idiot :|, thanks so much @stefan0xC !

EDIT:

Thanks, i can access the admin portal and all the settings are filled in properly! :)
